src/js/actions/tools.js

```diff
@@ -621,44 +644,53 @@ define(function (require, exports) {
                     currentTool.nativeToolName());
 
                 firstLaunch = false;
-                return Promise.join(initPromise, defaultPromise, resetPromise, dispatchPromise, policyPromise);
+                return Promise.join(initPromise, defaultPromise, resetPromise, dispatchPromise, policyPromise,
+                    removePromise);
             } else if (!vectorMaskMode) {
-                return Promise.join(initPromise, resetPromise, dispatchPromise, policyPromise);
+                return Promise.join(initPromise, resetPromise, dispatchPromise, policyPromise, removePromise);
             } else {
-                return Promise.join(initPromise, resetPromise, dispatchPromise, policyPromise)
+                return Promise.join(initPromise, resetPromise, dispatchPromise, policyPromise, removePromise)
                 .then(function () {
                     return UI.setSuppressTargetPaths(false);
                 })
                 .then(function () {
-                    return descriptor.playObject(vectorMaskLib.activateVectorMaskEditing());
+                    if (!currentLayer.vectorMaskEmpty) {
+                        return descriptor.playObject(vectorMaskLib.activateVectorMaskEditing());
+                    } else {
+                        return Promise.resolve();
+                    }
                 });
             }
         } else if (currentTool.id === "newSelect" || currentTool.id === "superselectVector") {
             if (!vectorMaskMode) {
                 return this.transfer(selectTool, toolStore.getToolByID("newSelect"))
                 .then(function () {
-                    return Promise.join(initPromise, dispatchPromise, policyPromise);
+                    return Promise.join(initPromise, dispatchPromise, policyPromise, removePromise);
                 });
             } else {
                 return this.transfer(selectTool, toolStore.getToolByID("superselectVector"))
                 .then(function () {
-                    return Promise.join(initPromise, dispatchPromise, policyPromise);
-                })
-                .then(function () {
-                    return descriptor.playObject(vectorMaskLib.activateVectorMaskEditing());
+                    return Promise.join(initPromise, dispatchPromise, policyPromise, removePromise);
                 })
                 .then(function () {
-                    descriptor.batchPlayObjects([vectorMaskLib.enterFreeTransformPathMode()],
-                        { synchronous: false });
+                    if (!currentLayer.vectorMaskEmpty) {
+                        return descriptor.playObject(vectorMaskLib.activateVectorMaskEditing())
+                            .then(function () {
+                                descriptor.batchPlayObjects([vectorMaskLib.enterFreeTransformPathMode()],
```

Comment:
not intentionally, but i'm not clear on how  synchronous: false affects promises.